WireAPI/Sources/WireAPI/Components/ResponseParser.swift

```diff
@@ -35,26 +40,44 @@ struct ResponseParser<Success> {
         parseBlocks = []
     }
 
+    /// Success with output data
+
     func success<Payload: Decodable & ToAPIModelConvertible>(
         code: Int,
         type: Payload.Type
     ) -> ResponseParser<Success> where Payload.APIModel == Success {
+        precondition(200 ..< 300 ~= code, "Requires a valid success code: 2xx")
```

Comment:
I think it would make sense to turn the code (Int) into an enum to ensure only success codes can be passed here. Same goes for failure. I will suggest these changes in my next PR.